DuckDuckGo/Subscription/PrivacyProDataReporter.swift

```diff
@@ -0,0 +1,289 @@
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+import BrowserServicesKit
+import DDGSync
+
+enum PrivacyProPromoParameters: String, CaseIterable {
+    case isReinstall
+    case fireButtonUser = "fireButtonUsed"
+    case syncUsed
+    case fireproofingUsed
+    case appOnboardingCompleted
+    case emailEnabled
+    case widgetAdded
+    case frequentUser
+    case longTermUser
+    case autofillUser
+    case validOpenTabsCount
+    case searchUser
+
+    /// Pick a randomized subset of parameters each time they are attached to a pixel
+    static func randomizedSubset(excluding excludedParameters: [PrivacyProPromoParameters] = []) -> [PrivacyProPromoParameters] {
+        let allParameters = Set(allCases).subtracting(Set(excludedParameters))
+        return Array(allParameters.shuffled().prefix(Int(allCases.count * 2/3)))
+    }
+}
+
+// swiftlint:disable identifier_name
+final class PrivacyProDataReporter {
+    enum Key {
+        static let fireCountKey = "com.duckduckgo.ios.privacypropromo.FireCount"
+        static let isWidgetAddedKey = "com.duckduckgo.ios.privacypropromo.WidgetAdded"
+        static let applicationLastSessionEndedKey = "com.duckduckgo.ios.privacypropromo.ApplicationLastSessionEnded"
+        static let searchCountKey = "com.duckduckgo.ios.privacypropromo.SearchCount"
+    }
+
+    enum Constants {
+        static let includedOrigins = "origins"
+    }
+
+    enum UseCase {
+        case messageID(String)
+        case origin(String?)
+        case debug
+    }
+
+    public static let shared = PrivacyProDataReporter()
+
+    private static let fireCountThreshold = 5
+    private static let frequentUserThreshold = 2
+    private static let longTermUserThreshold = 30
+    private static let autofillUserThreshold = 10
+    private static let openTabsCountThreshold = 3
+    private static let searchCountThreshold = 50
+
+    private lazy var includedOrigins = config.settings(for: .additionalCampaignPixelParams)[Constants.includedOrigins] as? [String]
+
+    private let config: PrivacyConfiguration
+    private let variantManager: VariantManager
+    private let userDefaults: UserDefaults
+    private let emailManager: EmailManager
+    private let tutorialSettings: TutorialSettings
+    private let appSettings: AppSettings
+    private let statisticsStore: StatisticsStore
+    private let secureVault: (any AutofillSecureVault)?
+    private var syncService: DDGSyncing?
+    private var tabsModel: TabsModel?
+    private let dateGenerator: () -> Date
+
+    init(config: PrivacyConfiguration = ContentBlocking.shared.privacyConfigurationManager.privacyConfig,
```

Comment:
The `PrivacyConfig` object is a struct, which means that we'll save a snapshot of it at initialization. If the Privacy Config is later updated, this object will not see those changes, so we should only refer to the manager and get the config dynamically each time.